pkg/util/crypt/crypt_dev.go

```diff
@@ -227,11 +215,11 @@ func copyDeviceContents(source, dest string, size int64) error {
 	for writtenSoFar < size {
 		numRead, err := syscall.Read(sourceFd, buffer)
 		if err != nil {
-			return fmt.Errorf("Unable to read the the file %s", source)
+			return fmt.Errorf("unable to read the the file: %s: %s", source, err)
 		}
 		numWritten, err := syscall.Write(destFd, buffer)
 		if err != nil {
-			return fmt.Errorf("Unable to write to destination %s", dest)
+			return fmt.Errorf("unable to write to destination: %s: %s", dest, err)
 		}
 
 		if numRead != numWritten {
```

Comment:
On line 226: `sylog.Debugf("numRead != numWritten")` I don't think on debugging logs we should output code conditions, a developer may not know what these variable mean. Better to log out the meaning of this condition, so an explanation like "the byte size read is different than the ones written", please reword as needed :+1:

Reply:
I already asked @schebro if that was necessary (https://github.com/sylabs/singularity/pull/3885#discussion_r301214749). Maybe I could change it  to something like this?

```go
sylog.Debugf("numRead = %v, numWritten = %v", numRead, numWrite)
```

Thoughts?

Reply:
better, but still doesn't explain what does `numRead` and `numWritten` variables even mean

Reply:
@WestleyK: @mem's 3930 fixed this part.